### PR TITLE
refactor Dockerfile

### PR DESCRIPTION
**Pull request by andrewhsu**
Refactor the `Dockerfile`:
* added `gcc` and `musl-dev` so `cgo` can operate
* update golang to latest 1.10.x
* multi-stage builds to have a minimal final image
* replace `docker-entrypoint.sh` (doesn't seem to be needed) with `gosec` binary

Final docker image built should be the same interface.
```
$ make image
Building the Docker image...
docker build -t securego/gosec:1.1.0-13-g6ca6e09 .
[+] Building 0.1s (14/14) FINISHED                                              
 => [internal] load Dockerfile                                             0.0s
 => => transferring dockerfile: 38B                                        0.0s
 => [internal] load .dockerignore                                          0.0s
 => => transferring context: 2B                                            0.0s
 => [internal] load metadata for docker.io/library/golang:1.10.4-alpine3.  0.0s
 => [internal] load metadata for docker.io/library/golang:1.10.4-alpine3.  0.0s
 => [internal] load build context                                          0.0s
 => => transferring context: 10.39kB                                       0.0s
 => [stage-1 1/3] FROM docker.io/library/golang:1.10.4-alpine3.8           0.0s
 => [internal] helper image for file operations                            0.0s
 => CACHED [stage-1 2/3] RUN apk add -U gcc musl-dev                       0.0s
 => CACHED [build 2/5] COPY . .                                            0.0s
 => CACHED [build 3/5] RUN apk add -U git make                             0.0s
 => CACHED [build 4/5] RUN go get -u github.com/golang/dep/cmd/dep         0.0s
 => CACHED [build 5/5] RUN make                                            0.0s
 => CACHED [stage-1 3/3] COPY --from=build /go/src/github.com/securego/go  0.0s
 => exporting to image                                                     0.0s
 => => exporting layers                                                    0.0s
 => => writing image sha256:2981dd166c325d76df902ea1087c0dad119b9d45af606  0.0s
 => => naming to docker.io/securego/gosec:1.1.0-13-g6ca6e09                0.0s
docker tag securego/gosec:1.1.0-13-g6ca6e09 securego/gosec:latest
touch image
$ make bootstrap
dep ensure
$ docker run --rm -it \
  -v "$GOPATH/src/github.com/securego/gosec:/go/src/github.com/securego/gosec" \
  -w /go/src/github.com/securego/gosec \
  securego/gosec:latest ./...
[gosec] 2018/09/25 04:28:27 including rules: default
[gosec] 2018/09/25 04:28:27 excluding rules: default
[gosec] 2018/09/25 04:28:27 Searching directory: /go/src/github.com/securego/gosec
...
Results:

Summary:
   Files: 44
   Lines: 5612
   Nosec: 16
  Issues: 0
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -183,7 +183,7 @@ You can run the `gosec` tool in a container against your local Go project. You j
 `GOPATH` of the container:
 
 ```
-docker run -it -v $GOPATH/src/<YOUR PROJECT PATH>:/go/src/<YOUR PROJECT PATH> securego/gosec /go/src/<YOUR PROJECT PATH>
+docker run -it -v $GOPATH/src/<YOUR PROJECT PATH>:/go/src/<YOUR PROJECT PATH> ./...
```
Comment (ccojocar) [inline README.md]: You missed the image from the command. I think it this remain the same `securego/gosec`.

File: Dockerfile [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
-ENV BIN=gosec
-ENV GOROOT=/usr/local/go
-ENV GOPATH=/go
-
-COPY $BIN /go/bin/$BIN
-COPY docker-entrypoint.sh /usr/local/bin
-
-ENTRYPOINT ["docker-entrypoint.sh"]
+FROM golang:1.10.4-alpine3.8
+RUN apk add -U gcc musl-dev
```
Comment (ccojocar) [inline Dockerfile]: `gosec` does not require `cgo` to operate. It works fine without and the image is `100MB` smaller. Please could you provide a some details about your use case which requires cgo?

Comment (andrewhsu) [reply Dockerfile]: fixed

File: README.md (hunk 1 above)
Comment (andrewhsu) [reply README.md]: fixed

File: Dockerfile (hunk 2 above)
Comment (andrewhsu) [reply Dockerfile]: Without `gcc` and `musl-dev`:
```
$ docker run --rm -t -v `pwd`:/go/src/github.com/opencontainers/runc -w /go/src/github.com/opencontainers/runc securego/gosec -include G401 -quiet ./...
exec: "gcc": executable file not found in $PATH
cgo failed: [go tool cgo -objdir /tmp/github.com_opencontainers_runc_libcontainer_system_C530284581 -- -I /tmp/github.com_opencontainers_runc_libcontainer_system_C530284581 sysconfig.go]: exit status 1
...
```

With `gcc` and `musl-dev`:
```
$ docker run --rm -t -v `pwd`:/go/src/github.com/opencontainers/runc -w /go/src/github.com/opencontainers/runc securego/gosec -include G401 -quiet ./...
```

**Pull request merged**